widgetssdk/src/main/java/com/glia/widgets/callvisualizer/ActivityWatcherController.kt

```diff
@@ -0,0 +1,210 @@
+package com.glia.widgets.callvisualizer
+
+import android.app.Activity
+import android.content.Intent
+import android.media.projection.MediaProjectionManager
+import android.view.View
+import androidx.activity.ComponentActivity
+import androidx.activity.result.ActivityResultLauncher
+import androidx.annotation.VisibleForTesting
+import com.glia.androidsdk.Glia
+import com.glia.androidsdk.GliaException
+import com.glia.widgets.R
+import com.glia.widgets.callvisualizer.controller.CallVisualizerController
+import com.glia.widgets.chat.ChatView
+import com.glia.widgets.core.dialog.Dialog.*
+import com.glia.widgets.core.dialog.DialogController
+import com.glia.widgets.core.dialog.model.DialogState
+import com.glia.widgets.core.dialog.model.DialogState.MediaUpgrade
+import com.glia.widgets.core.screensharing.ScreenSharingController
+import com.glia.widgets.filepreview.ui.FilePreviewView
+import com.glia.widgets.helper.Logger
+import com.glia.widgets.view.head.controller.ServiceChatHeadController
+import com.glia.widgets.view.unifiedui.exstensions.prepareContextWithStyle
+import java.lang.ref.WeakReference
+
+class ActivityWatcherController(
+    private val callVisualizerController: CallVisualizerController,
+    private val screenSharingController: ScreenSharingController,
+    private var serviceChatHeadController: ServiceChatHeadController
+) : ActivityWatcherContract.Controller {
+
+    companion object {
+        private val TAG = ActivityWatcherController::class.java.simpleName
+    }
+
+    @Mode
+    private var dialogType: Int = MODE_NONE
```

Comment:
NIT: Maybe can be renamed to `currentDialogType`

Reply:
done